File: src/main/java/no/nav/domain/pensjon/regler/grunnlag/InstOpphPeriodegrunnlag.java
package no.nav.domain.pensjon.regler.grunnlag;

import no.nav.domain.pensjon.regler.kode.OppholdTypeCti;
import no.nav.domain.pensjon.regler.kode.VarighetTypeCti;

import java.io.Serializable;
import java.util.Date;

/**
 * En periode for et opphold på institusjon
 */
public class InstOpphPeriodegrunnlag implements Serializable {
    /**
     * Kun brukt i PEN
     */
    private long instOppholdId;

    /**
     * Kode som angir type opphold.
     */
    private OppholdTypeCti oppholdType;

    /**
     * Varighetstypen på institusjonsoppholdet.
     */
    private VarighetTypeCti varighetType;
    /**
     * Fradato for oppholdet
     */
    private Date fom;
    /**
     * Tildato for oppholdet
     */
    private Date tom;
    /**
     * Forventet tildato for oppholdet - oppholdet pågår.
     */
    private Date forventetTom;
    /**
     * Kun brukt i PEN
     */
    private String registerOpprettetAv;
    /**
     * Kun brukt i PEN
     */
    private String registerEndretAv;
    /**
     * Kun brukt i PEN
     */
    private Date registerOpprettetDato;
    /**
     * Kun brukt i PEN
     */
    private Date registerEndretDato;
    /**
     * Kun brukt i PEN
     */
    private String registerKilde;
    /**
     * Angir om institusjonsoppholdet er overført til en annen institusjon.
     */
    private boolean overfort;
    /**
     * Kun brukt i PEN
     */
    private boolean bruk;
}

File: src/main/kotlin/no/nav/pensjon/regler/domain/grunnlag/InstOpphPeriodegrunnlag.kt
package no.nav.pensjon.regler.domain.grunnlag

import no.nav.pensjon.regler.domain.kode.OppholdTypeCti
import no.nav.pensjon.regler.domain.kode.VarighetTypeCti
import java.io.Serializable
import java.util.*

/**
 * En periode for et opphold på institusjon
 */
@Deprecated("Ikke i bruk")
class InstOpphPeriodegrunnlag : Serializable {
    /**
     * Kun brukt i PEN
     */
    var instOppholdId: Long = 0

    /**
     * Kode som angir type opphold.
     */
    var oppholdType: OppholdTypeCti? = null
    // TODO Hva skal her? Vi har ingen Enum for dette, Pesys har InstitusjonsoppholdTypeCode med AS (alder og sykehjem),FO (fengsel),HS (helseinstitusjon)
//    var oppholdTypeEnum: OppholdTypeCti? = null

    /**
     * Varighetstypen på institusjonsoppholdet.
     */
    var varighetType: VarighetTypeCti? = null
    // TODO Finner ingen koder for dette hverken i regler eller pesys
//    var varighetTypeEnum: Varigh? = null

    /**
     * Fradato for oppholdet
     */
    var fom: Date? = null

    /**
     * Tildato for oppholdet
     */
    var tom: Date? = null

    /**
     * Forventet tildato for oppholdet - oppholdet pågår.
     */
    var forventetTom: Date? = null

    /**
     * Kun brukt i PEN
     */
    var registerOpprettetAv: String? = null

    /**
     * Kun brukt i PEN
     */
    var registerEndretAv: String? = null

    /**
     * Kun brukt i PEN
     */
    var registerOpprettetDato: Date? = null

    /**
     * Kun brukt i PEN
     */
    var registerEndretDato: Date? = null

    /**
     * Kun brukt i PEN
     */
    var registerKilde: String? = null

    /**
     * Angir om institusjonsoppholdet er overfårt til en annen institusjon.
     */
    var overfort = false

    /**
     * Kun brukt i PEN
     */
    var bruk: Boolean = true

}